src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java

```diff
@@ -50,31 +50,38 @@
     /**
      * The Constructor of GraphClient.
      *
-     * @param host      The host of graph services.
-     * @param port      The port of graph services.
-     * @param timeout   The timeout of RPC request.
-     * @param retry     The number of retries when connection failure.
+     * @param addresses       The addresses of graph services.
+     * @param timeout         The timeout of RPC request.
+     * @param connectionRetry The number of retries when connection failure.
+     * @param executionRetry  The number of retries when execution failure.
      */
-    public GraphClientImpl(String host, int port, int timeout, int retry) {
+    public GraphClientImpl(List<HostAndPort> addresses, int timeout, int connectionRetry,
+                           int executionRetry) {
         checkArgument(timeout > 0);
-        checkArgument(retry > 0);
-        if (!InetAddresses.isInetAddress(host) || (port <= 0 || port >= 65535)) {
-            throw new IllegalArgumentException(String.format("%s:%d is not a valid address", host, port));
-        }
+        checkArgument(connectionRetry > 0);
+        addresses.forEach(address -> {
+            String host = address.getHost();
+            int port = address.getPort();
+            if (!InetAddresses.isInetAddress(host) || (port <= 0 || port >= 65535)) {
```

Comment:
the handle to check address and port may be better put in  `fetchOptionValue`

Reply:
`fetchOptionValue` is in Importer and this is `GraphClient`